inginious/frontend/pages/course_admin/api_auth.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+
+from flask import request, Response, abort
+import jwt
+from datetime import datetime
+import json
+
+from inginious.frontend.pages.utils import INGIniousPage
+
+
+class DataAPIPage(INGIniousPage):
+    """ Token verification and handler for the data API """
+
+    def __init__(self):
+        super().__init__()
+        self.jwt_key = str(self.app.jwt_key)
+
+    def verify(self):
+        """ verify if a token is valid """
+        auth_header = request.headers.get('Authorization')
+        try:
+            token = auth_header.split()[1]
+            decoded = jwt.decode(token, self.jwt_key, algorithms=["HS256"])
+        except:
+            abort(401, description="No authentification token found.")
+        try:
+            tok_id = decoded["_id"]
+            expire = datetime.strptime(decoded["expire"], "%Y-%m-%d %H:%M:%S")
+            courseid = decoded["courseid"]
+        except:
+            abort(401, description="Invalid token.")
+
+        stored_token = self.database.tokens.find_one({"$expr": {"$eq": ["$_id", {"$toObjectId": tok_id}]}})
```

Comment:
Geee... such a complex request to fetch an element by its ObjectId ?? Import and use `bson.ObjectId` to perform a simple `_id` match request.